docs/rest_api_guide/02_RESTauthentication.rst

```diff
@@ -0,0 +1,365 @@
+
+The response payload is a JSON document containing the username of the newly created user.
+
+**Errors**
+
+The following error responses may be returned
+
+.. list-table::
+    :widths: 20 50
+    :header-rows: 1
+
+    * - HTTP Code
+      - Reason
+    * - 400
+      - Incomplete or badly formed request payload
+    * - 403
+      - A user without admin permissions tried to add a new user
+    * - 409
+      - The username is already in use
+
+
+**Example**
+
+.. code-block:: console
+
+   curl -X POST /fledge/user -d'
+   {
+    "username"    : "david",
+    "password"    : "1nv1nc1ble",
+    "permissions" : "admin",
+    "realname"    : "David Brent"
+   }
+
+Get All Users
+-------------
+
+``GET /fledge/user`` - Retrieve data on all users
+
+**Response Payload**
+
+A JSON document which all users in a JSON array.
+
+.. list-table::
+    :widths: 20 20 50 30
+    :header-rows: 1
+
+    * - Name
+      - Type
+      - Description
+      - Example
+    * - [].username
+      - string
+      - The username of the new user to add
+      - david
+    * - [].permissions
+      - string
+      - The permissions that new user should be given
+      - admin
+    * - [].realname
+      - string
+      - The real name of the user. This is used for display purposes only.
+      - David Brent
+
+.. note::
+
+   This payload does not include the password of the user.
+
+**Example**
+
+.. code-block:: console
+
+   curl -X GET /fledge/user
+
+
+Returns the response payload
+
+.. code-block:: console
+    {
+        "users" : [
+                    {
+                       "username"    : "david",
+                       "permissions" : "admin",
+                       "realname"    : "David Brent"
+                    },
+                    {
+                       "username"    : "paul",
+                       "permissions" : "user",
+                       "realname"    : "Paul Smith"
+                    }
+                  ]
+    }
+
+
+
+Get User
+--------
+
+``GET /fledge/user/{username}`` - Retrieve data on a user
+
+**Response Payload**
+
+A JSON document which describes the user.
+
+.. list-table::
+    :widths: 20 50
+    :header-rows: 1
+
+    * - Name
+      - Type
+      - Description
+      - Example
+    * - username
+      - string
+      - The username of the new user to add
+      - david
+    * - permissions
+      - string
+      - The permissions that new user should be given
+      - admin
+    * - realname
+      - string
+      - The real name of the user. This is used for display purposes only.
+      - David Brent
+
+
+..note::
+
+    This payload does not include the password of the user.
+
+**Example**
+
+.. code-block:: console
+
+  GET /fledge/user/david**
+
+Returns the response payload
+
+.. code-block:: console
+
+    {
+        "username"    : "david",
+        "permissions" : "admin",
+        "realname"    : "David Brent"
+    }
+
+Update User
+-----------
+
+``PUT /fledge/user/{username}`` - update a user
+
```

Comment:
We have two routes to update the user.
For normal user
```
             curl -H "authorization: <token>" -X PUT -d '{"real_name": "AJ"}' http://localhost:8081/fledge/user

```

For admin user

```
            curl -H "authorization: <token>" -X PUT -d '{"description": "A new user", "real_name": "Admin", "access_method": "pwd"}' http://localhost:8081/fledge/admin/{user_id}

```


See - FOGL-1226 JIRA to support more fields for profile update of user